### PR TITLE
Add Track::IsFilterable() method to allow non-filterable tracks

### PR DESCRIPTION
**Pull request by pierricgimmig**
Scheduler track is now always visible (b/181671054). 
"Pinned" tracks are also not be filterable.

### CODE REVIEW

**Review by reichlfl — COMMENTED** [review]
C++ part LGTM, but this will break some E2E tests that use the track filtering - at least orbit_create_frame_track, and it will make the results from orbit_tracks less meaningful.

I'd suggest to change the create_frame_track test to expect the scheduler to be visible - filter_string should already work with a list of names, so you should be able to include the scheduler there. If not, you can split the FilterTracks test to not verify the # of tracks, and use MatchTracks yourself.

For orbit_tracks, at least increase the expected # of tracks where needed.

File: src/OrbitGl/TrackManager.cpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ void TrackManager::UpdateFilteredTrackList() {
   for (const auto& track : sorted_tracks_) {
     std::string lower_case_label = absl::AsciiStrToLower(track->GetLabel());
     for (auto& filter : filters) {
-      if (absl::StrContains(lower_case_label, filter)) {
+      if (!track->IsFilterable() || absl::StrContains(lower_case_label, filter)) {
```
Comment (karupayun) [inline src/OrbitGl/TrackManager.cpp:166]: Nit: consider moving this if to the track-loop.

**Review by karupayun — APPROVED** [review]
LGTM Pierric! Thanks for solving it!

File: src/OrbitGl/TrackManager.cpp (hunk 1 above)
Comment (pierricgimmig) [reply src/OrbitGl/TrackManager.cpp:166]: Oh yes, of course, thanks!

**Comment by pierricgimmig** [comment]
> but this will break some E2E tests

Arrgh, I will implement your suggestions, thanks @reichlfl

**Pull request closed without merging**